core/src/env.rs

```diff
@@ -185,21 +184,65 @@ impl<E: Ext> LaterExt<E> {
         res
     }
 
-    /// Unset inner ext
-    pub fn take(self) -> Option<E> {
-        self.inner.borrow_mut().take()
+    /// Creates clone for the current reference.
+    ///
+    /// Clone type differs from the [`ExtCarrier`]. For rationale see [`ExtCarrierClone`] docs.
+    pub fn create_carrier_clone(&self) -> ExtCarrierClone<E> {
```

Comment:
Maybe to rename it to `cloned` (allusion to `Option::cloned()` which clones the inner value)?

```suggestion
    pub fn cloned(&self) -> ClonedExtCarrier<E> {
```

Reply:
done in [0817d34](https://github.com/gear-tech/gear/pull/853/commits/0817d34d8fe8168d3d646a8c27dd205e507d585b)